src/api/app/views/webui2/webui/repositories/distributions.html.haml

```diff
@@ -0,0 +1,54 @@
+- @pagetitle = "Add Repository to #{@project}"
+- @metarobots = 'noindex'
+
+.card.mb-3
+  = render partial: 'webui/project/tabs', locals: { project: @project }
+  .card-body
+    %h3.mb-3 Add Repositories to #{@project}
+    - @distributions.each do |vendor, list|
+      %h6.pt-3
+        - if resolve_asset_path("icons/distributions-#{vendor.downcase}")
+          = image_tag("icons/distributions-#{vendor.downcase}")
+        #{vendor} distributions
+      .form-group.pl-3
+        - list.each do |distribution|
+          - selected = @project.has_distribution(distribution['project'], distribution['repository'])
+          .custom-control.custom-checkbox.custom-control-inline
+            = check_box_tag 'distributions[]', distribution['reponame'], selected,
+                                               id: "repo_#{replace_jquery_meta_characters(distribution['reponame'])}",
+                                               class: 'repocheckbox custom-control-input'
+            = label_tag :distribution_name, for: "repo_#{replace_jquery_meta_characters(distribution['reponame'])}", class: 'custom-control-label' do
+              = distribution['name']
+        - list.each do |distribution|
+          = form_tag({ action: :create, project: @project }, remote: true, class: 'd-none',
```

Comment:
@dmarcoux I don't think this is a blocker.

Regarding the actual issue. I think it would be nice if we could use path helpers here, but I also understand @Ana06 argument. Essentially we should make the repo distribution page resourceful and then we would have the path helpers we lack right now.

Reply:
Fine